websites/Y/Yandex/presence.ts

```diff
@@ -0,0 +1,146 @@
+const presence = new Presence({
+		clientId: "1035433850593095740",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	homepageInput =
+		document.querySelector<HTMLInputElement>('input[name="text"]'),
+	{ host, pathname } = document.location;
+
+enum Assets {
+	Cover = "https://i.imgur.com/SFXKtLF.png",
+	Search = "https://i.imgur.com/wYVlwJX.png",
+	View = "https://i.imgur.com/hxvvGUi.png",
+}
+
+async function getStrings() {
+	return presence.getStrings(
+		{
+			search: "general.search",
+			view: "general.viewing",
+			viewHome: "general.viewHome",
+			searchFor: "general.searchFor",
+			searchSomething: "general.searchSomething",
+			watching: "general.watching",
+		},
+		await presence.getSetting<string>("lang").catch(() => "ru")
+	);
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
+
+function textContent(tags: string) {
+	return document.querySelector(tags)?.textContent;
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Cover,
+			startTimestamp: browsingTimestamp,
+		},
+		[newLang, privacy, logo, time] = await Promise.all([
+			presence.getSetting<string>("lang").catch(() => "ru"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("logo"),
+			presence.getSetting<boolean>("time"),
+		]);
+
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
+	switch (host) {
+		case "ya.ru":
+			presenceData.details = strings.viewHome;
+			if (homepageInput?.value !== "") {
```

Comment:
An empty string is falsy
```suggestion
			if (homepageInput?.value) {
```